components/eamxx/CMakeLists.txt

```diff
@@ -51,7 +51,7 @@ endif()
 # to be on. For now, simply ensure Kokkos Serial is enabled
 option (Kokkos_ENABLE_SERIAL "" ON)
 
-set (CMAKE_CXX_STANDARD 14)
+set (CMAKE_CXX_STANDARD 17)
```

Comment:
Nice catch, @bartgol ! (I forgot to add you to the reviewer list.) We actually do use some very very basic C++17 features, and I believe they're related to Kokkos, but I'd have to review. If this causes problems, we can probably revert to C++14 and work around them in our feature branches, but it would be nice if we could switch without incident.

Reply:
I don't know if it will cause issues. My main concern is to be able to compile correctly on all target machines. I think they should all be C++17 compliant by now, or else Kokkos would not be forcing C++17, I think. The caveat is to make sure the chain of TPL libs we need on those machines is fully supported with the C++17 compliant compiler suite.

Reply:
I'll go along with whatever the EAMxx team thinks is best. Let me know if you'd like this reverted back to C++14, or whether I can help determine whether C++17 is practical.